Automattic/wp-calypso
# Stats: Some slider-related cleanup

Related to #84859.

## Proposed Changes

* Removes old `TierUpgradeSlider` implementation.
* Removes no longer used code paths.
* Removes some test data.
* Adds some comments on things that might need further testing.

## Testing Instructions



* Launch the Calypso Live link.
* Navigate to **Stats → Traffic**.
* Click on CTA to purchase Stats plan.
* Add `stats/tier-upgrade-slider` to URL flags.
* Confirm new tier slider is visible and updates the page correctly.
* Click the "choose a commercial plan" link in the highlighted box.
* Confirm the new tier slider shows correctly here too.

Should work correctly on both DotCom and Jetpack sites.

## Pre-merge Checklist
